### DISCUSSION TITLE
Is it possible to write a policy to check a crud action against the authed user's email address?

### DISCUSSION THREAD

**Discussion by MincePie** [supabase/supabase#16629] [Questions]
I am trying to figure out how to write a supabase RLS policy that allows an authed user with a specific email address to delete any post.

The auth users table has an email attribute. I want that attribute to be the permissive policy on a delete function on another table in the public schema. 

I've tried a lot of different things in the posts table delete RLS policy, along the lines of: 

`auth.email() = user_email(hello@myemaildomain.com)
`
They all generate an error saying this isn't a valid policy.

Does anyone know how to make this work. My only authentication method is email and password. My users table has an email field. I want the user with the email address I specify in this policy to be able to delete any post. How can I write that?

Chat gpt thinks the answer is:

```
CREATE OR REPLACE FUNCTION is_allowed_to_delete_post()
  RETURNS boolean AS $$
DECLARE
  current_user_email text;
BEGIN
  SELECT email INTO current_user_email FROM users WHERE id = current_setting('jwt.claims.user_id')::uuid;
  RETURN current_user_email = 'hello@hello.com';
END;
$$ LANGUAGE plpgsql;

-- Create a row-level security policy on the "posts" table
CREATE POLICY allow_delete_policy
  FOR DELETE ON posts
  USING (is_allowed_to_delete_post());

```

I can't find any examples in the supabase docs showing how to incorporate this approach.

**Comment by GaryAustin1** [comment]
If you are wanting to compare a specific email string against the current users then
`auth.jwt() ->> 'email' = 'myemail@email.com' `
Should work.  (note auth.email() = should also work, but is deprecated).
https://supabase.com/docs/guides/auth/row-level-security#verifying-email-domains

**Reply by MincePie** [reply]
thank you -that worked (i think)

**Reply by GaryAustin1** [reply]
You should not add new answers and comment on the current one if it applies.
You don't really show your RLS policies, just the one part of it that looks right at a quick glance.

If you are inserting you need to have a policy for that it involves WITH CHECK and not USING.

Also it is possible your jwt/access token is not really set when you are doing the call.

**Reply by MincePie** [reply]
I am sorry. I wrote the first answer when I managed to get supabase to accept the polilcy that I had written ( a single line in the format you suggested). I then tried to use that permission (by posting to that table, and am not able to do so), so I went back to check the policy, and found that supabase had modified the format after I had saved it. I don't know what check and using mean. I'll try to find a tutorial that might help me learn what those terms mean in the context of this issue. thanks anyway for trying to help.

**Reply by GaryAustin1** [reply]
Not sure if you are using the UI for RLS or SQL.
![image](https://github.com/supabase/supabase/assets/54564956/ab481c75-cc8e-4304-9fa4-223751810270)
![image](https://github.com/supabase/supabase/assets/54564956/ff4664d4-1f75-47d0-9770-343ca700c425)
![image](https://github.com/supabase/supabase/assets/54564956/8bcf7313-9ed0-482d-8226-ea5b66253616)
Show the UI for select and insert.  Not the WITH CHECK in the insert policy.  That tells the RLS to look at the new record.  USING tells RLS to look at the existing record.

**Comment by MincePie** [comment]
I tried this.

I noticed that when I save the policy and go back and have a look, supabase has modified it to:

((auth.jwt() ->> 'email'::text) = 'hello@hello.com'::text)

I also cannot actual post data to this table, even when im logged in as the user with this email address.

the code im working with is this starter tutorial: https://supabase.com/docs/guides/auth/auth-helpers/nextjs. I haven't done anything other than try to make a policy that is managed with this RLS approach.
